### PR TITLE
Fix check_fit_non_negative failing estimator check (+3 others)

### PR DESCRIPTION
**Pull request by TamaraAtanasoska**
## Description
This PR fixes the following failing estimator checks:
- `check_fit_non_negative`
-  `check_regressor_data_not_an_array`
- `check_estimators_fit_returns_self`
- `check_classifier_data_not_an_array`

I discovered some of these by looking at the xpassed as side effects to fixing others, so some might overlap with what #1397  has listed. No major changes in this PR.

@adrinjalali 

## Tests

- [ ] no new tests required
- [ ] new tests added
- [x] existing tests adjusted

## Documentation

- [x] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

File: fairlearn/adversarial/_adversarial_mitigation.py [hunk 1]
```diff
@@ -445,7 +444,7 @@ def fit(self, X, y, *, sensitive_features=None):
             training data.
         """
         X, Y, A = self._validate_input(X, y, sensitive_features, reinitialize=True)
```
Comment (adrinjalali) [inline fairlearn/adversarial/_adversarial_mitigation.py:446]: I think we shoulddn't introduce a new `Y` here, this seems to need to be 

```suggestion
        X, y, A = self._validate_input(X, y, sensitive_features, reinitialize=True)
```

**Review by adrinjalali — COMMENTED** [review]
Otherwise LGTM.

Also, merge conflicts.

File: fairlearn/adversarial/_adversarial_mitigation.py (hunk 1 above)
Comment (TamaraAtanasoska) [reply fairlearn/adversarial/_adversarial_mitigation.py]: I agree about the consistency! Unfortunately it is everywhere a capital `Y` jus in the function signature a small `y`. I capitalised it in the signature. [6fb9423](https://github.com/fairlearn/fairlearn/pull/1398/commits/6fb9423e1e0608ad75afa771ab78964d60346c8f)

**Comment by TamaraAtanasoska** [comment]
@adrinjalali just pinging to add this to your notifications now after all the updates 🧹

File: fairlearn/adversarial/_adversarial_mitigation.py (hunk 1 above)
Comment (adrinjalali) [reply fairlearn/adversarial/_adversarial_mitigation.py:451]: by convention on the sklearn side we always have `y`, except for historical reasons where it was `Y` before everything became `y`. So I'd go with `y` here and deprecate `Y`.

**Review by adrinjalali — APPROVED** [review]

**Pull request merged**